tls/s2n_handshake_io.c

```diff
@@ -1285,7 +1285,8 @@ static int s2n_handshake_read_io(struct s2n_connection *conn)
          */
         if (message_type == TLS_HELLO_REQUEST) {
             POSIX_GUARD(s2n_client_hello_request_recv(conn));
-            return S2N_SUCCESS;
+            POSIX_GUARD(s2n_stuffer_wipe(&conn->handshake.io));
```

Comment:
the while loop is on `conn->in` but we are wiping `handshake.io`. What is the relation?

Reply:
Understandable question, this method is not the most readable. We initially read records into conn->in. Records potentially contain multiple messages of the same type. If it's a handshake type record, then we then [read each message into handshake.io](https://github.com/aws/s2n-tls/blob/2f34016aba8599fb30b617c8446b1833a4adf68f/tls/s2n_handshake_io.c#L1025), which is where all our message handlers know to look for it.